File: todo-list/src/ToDoListController.java
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

import static java.lang.Integer.parseInt;

public class ToDoListController {
    private ToDoListView view;
    private ToDoListModel model;
    private boolean enMenuPrincipal = true;

    public ToDoListController(ToDoListView view, ToDoListModel model) {
        this.view = view;
        this.model = model;

        // CARGAR LISTENERS
        view.cargarAgregarItemListener(new AgregarItemListener());
        view.cargarMostrarCompletadosListener(new MostrarCompletadosListener());
        view.cargarSalirListener(new SalirListener());
        view.addDeleteButtonListener(new DeleteItemListener());
        view.addFinishButtonListener(new FinishItemListener());

        // MOSTRAR MENU PRINCIPAL AL INICIO & CONFIGURACIÓN INICIAL
        view.updateUnfinished(model.getUnfinishedItems());
        view.actualizarNombreBotonMenu(enMenuPrincipal);
        view.ocultarAgregarItem(!enMenuPrincipal);
    }


    class AgregarItemListener implements ActionListener {
        public void actionPerformed(ActionEvent e) {
            String description = view.showInputDialog("Ingrese la descripción del nuevo ítem:");
            if (description != null && !description.isEmpty()) {
                model.addItem(new Item(description));
                if (enMenuPrincipal) {
                    view.updateUnfinished(model.getUnfinishedItems());
                } else {
                    view.updateCompleted(model.getCompletedItems());
                }
                view.limpiarTextArea();
            } else {
                view.mostrarError("El campo de descripción está vacío.");
            }
        }
    }

    class MostrarCompletadosListener implements ActionListener {
        public void actionPerformed(ActionEvent e) {
            if (enMenuPrincipal) {
                view.updateCompleted(model.getCompletedItems());
            } else {
                view.updateUnfinished(model.getUnfinishedItems());
            }
            enMenuPrincipal = !enMenuPrincipal;
            view.actualizarNombreBotonMenu(enMenuPrincipal);
            view.ocultarAgregarItem(!enMenuPrincipal);
        }
    }

    class SalirListener implements ActionListener {
        public void actionPerformed(ActionEvent e) {
            System.exit(0);
        }
    }

    class DeleteItemListener implements ActionListener {
        public void actionPerformed(ActionEvent e) {
            System.out.println("Delete button clicked"); // Mensaje de depuración
            String command = e.getActionCommand();
            System.out.println("Command: " + command); // Mensaje de depuración
            int itemId = parseInt(command);
            System.out.println("Item ID: " + itemId); // Mensaje de depuración
            model.deleteItem(model.getItem(itemId)); // Eliminar el ítem del modelo
            if (enMenuPrincipal) {
                view.updateUnfinished(model.getUnfinishedItems()); // Actualizar vista
            } else {
                view.updateCompleted(model.getCompletedItems()); // Actualizar vista
            }
        }
    }


    class FinishItemListener implements ActionListener {
        public void actionPerformed(ActionEvent e) {
            System.out.println("FINISH CLICKED");
        }
    }
}

File: todo-list-py/src/todoListController.py
# todoListController.py
from pubsub import pub
from item import Item


class ToDoListController:
    def __init__(self, view, model):
        self.view = view
        self.model = model
        self.en_menu_principal = True

        pub.subscribe(self.agregar_item, "add_item")
        pub.subscribe(self.mostrar_completados, "mostrar_completados")
        pub.subscribe(self.salir, "salir")
        pub.subscribe(self.finish_item, "finish_item")
        pub.subscribe(self.delete_item, "delete_item")
        pub.subscribe(self.update_unfinished_items, "request_unfinished_items")
        pub.subscribe(self.update_completed_items, "request_completed_items")

    def agregar_item(self, description):
        self.model.add_item(Item(description))
        self.update_view()

    def mostrar_completados(self):
        self.en_menu_principal = not self.en_menu_principal
        self.update_view()

    def update_view(self):
        if self.en_menu_principal:
            unfinished_items = self.model.get_unfinished_items()
            self.view.update_unfinished(unfinished_items)
        else:
            completed_items = self.model.get_completed_items()
            self.view.update_unfinished(completed_items)  # Aquí actualizamos con los completados
        self.view.actualizar_nombre_boton_menu(self.en_menu_principal)
        self.view.ocultar_agregar_item(not self.en_menu_principal)

    def salir(self):
        exit(0)

    def delete_item(self, item_id):
        self.model.delete_item(self.model.get_item(int(item_id)))
        self.update_view()

    def finish_item(self, item_id):
        self.model.finish_item(self.model.get_item(int(item_id)))
        self.update_view()

    def update_unfinished_items(self):
        self.update_view()  # No necesitamos obtener los elementos no finalizados aquí

    def update_completed_items(self):
        self.update_view()  # No necesitamos obtener los elementos completados aquí
